### PR TITLE
Introduce AttachmentListItem wrapper class

### PR DESCRIPTION
**Pull request by zsmb13**
Wraps the `Attachment` network model into the `AttachmentListItem` class, so that it's not used directly in `AttachmentListItemAdapter`. This allows us to add extra properties to the list item later, if needed.

See previous discussion in https://github.com/GetStream/stream-chat-android/pull/614#discussion_r484452890

### CODE REVIEW

File: library/src/main/java/com/getstream/sdk/chat/adapter/AttachmentViewHolder.java [hunk 1]
```diff
@@ -69,7 +69,7 @@ public void bind(@NonNull Context context,
         this.context = context;
         this.messageListItem = messageListItem;
         this.message = message;
-        this.attachment = attachment;
+        this.attachment = attachmentItem.getAttachment();
```
Comment (zsmb13) [inline library/src/main/java/com/getstream/sdk/chat/adapter/AttachmentViewHolder.java:72]: Decided to not change the type of the private fields inside the ViewHolder classes, for now. This can be changed later so that the entire item is stored if needed, but that's all private implementation details, and it doesn't break the public API.

File: library/src/main/java/com/getstream/sdk/chat/adapter/AttachmentListItemAdapter.java [hunk 2]
```diff
@@ -22,27 +22,29 @@
...
-    public AttachmentListItemAdapter(Context context, @NonNull MessageListItem.MessageItem messageListItem, @NonNull MessageViewHolderFactory factory) {
+    public AttachmentListItemAdapter(Context context,
+                                     @NonNull MessageListItem.MessageItem messageListItem,
+                                     @NonNull MessageViewHolderFactory factory) {
         this.context = context;
         this.messageListItem = messageListItem;
         this.message = messageListItem.getMessage();
         this.factory = factory;
-        this.attachments = message.getAttachments();
+        this.attachments = CollectionsKt.map(message.getAttachments(), AttachmentListItem::new);
```
Comment (zsmb13) [inline library/src/main/java/com/getstream/sdk/chat/adapter/AttachmentListItemAdapter.java:40]: Example of how handy Kotlin collection functions are, even when writing Java code

**Review by ogkuzmin — APPROVED** [review]

**Review by adasiewiczr — APPROVED** [review]

File: library/src/main/java/com/getstream/sdk/chat/adapter/AttachmentListItemAdapter.java (hunk 2 above)
Comment (JcMinarro) [reply library/src/main/java/com/getstream/sdk/chat/adapter/AttachmentListItemAdapter.java:40]: I have never used it from Java, Taking note ;)

File: library/src/main/java/com/getstream/sdk/chat/adapter/MessageViewHolderFactory.kt [hunk 3]
```diff
@@ -37,9 +36,9 @@ open class MessageViewHolderFactory {
     }
 
     open fun getAttachmentViewType(
-        attachment: Attachment
+        attachmentItem: AttachmentListItem
     ): Int {
-        return when (attachment.type) {
+        return when (attachmentItem.attachment.type) {
```
Comment (JcMinarro) [inline library/src/main/java/com/getstream/sdk/chat/adapter/MessageViewHolderFactory.kt:41]: It seems that `AttachmentListItem` could be a `sealed class`, so it will be easier to handle multiple types.
It could be covered on another PR when `ViewHolder` and `Adapter` will refactored

**Review by JcMinarro — DISMISSED** [review]

File: library/src/main/java/com/getstream/sdk/chat/adapter/MessageViewHolderFactory.kt (hunk 3 above)
Comment (zsmb13) [reply library/src/main/java/com/getstream/sdk/chat/adapter/MessageViewHolderFactory.kt:41]: I'd leave this for later, the wrapper feels like enough preparation for future expansion for now

**Comment by JcMinarro** [comment]
BTW, @zsmb13 could you update our `CHANGELOG.md` with the new updates?

**Review by JcMinarro — APPROVED** [review]
